Libraries/LibGUI/TextBox.cpp

```diff
@@ -37,4 +37,14 @@ TextBox::~TextBox()
 {
 }
 
+void TextBox::keydown_event(GUI::KeyEvent& event)
+{
+    if (event.key() == Key_Up)
+        on_up_pressed();
+    else if (event.key() == Key_Down)
+        on_down_pressed();
```

Comment:
We have to null-check the `Function` objects before calling them:

```cpp
if (on_up_pressed)
    on_up_pressed();
```

Reply:
Ah of course, copy-paste strikes again :)